src/snowflake/snowpark/dataframe.py

```diff
@@ -1075,13 +1075,24 @@ def to_pandas(
                 **kwargs,
             )
 
-        # if the returned result is not a pandas dataframe, raise Exception
-        # this might happen when calling this method with non-select commands
-        # e.g., session.sql("create ...").to_pandas()
         if block:
             if not isinstance(result, pandas.DataFrame):
+                query = self._plan.queries[-1].sql.strip().lower()
+                is_select_statement = is_sql_select_statement(query)
+                if is_select_statement:
+                    _logger.warning(
+                        "to_pandas() did not return a pandas dataframe. "
+                        "If the query result format is set to JSON, accuracy of returned data is not guaranteed. "
+                        "Please set query result format to ARROW for best accuracy."
```

Comment:
I would be more vague here since we haven't really tested the accuracy issue.
also `to_pandas() did not return a pandas dataframe.` is a bit misleading, to user, the `to_pandas` is still returning a pandas DF

```suggestion
"The query result format is set to JSON, the resulting pandas DataFrame may not align with the result returned in the ARROW format. For best compatibility with to_pandas(), set the query result format to ARROW."
```